sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java

```diff
@@ -73,109 +80,182 @@ public static BeamSqlEnv inMemory(TableProvider... tableProviders) {
     return withTableProvider(inMemoryMetaStore);
   }
 
-  private void registerBuiltinUdf(Map<String, List<Method>> methods) {
-    for (Map.Entry<String, List<Method>> entry : methods.entrySet()) {
-      for (Method method : entry.getValue()) {
-        connection.getCurrentSchemaPlus().add(entry.getKey(), UdfImpl.create(method));
-      }
-    }
+  public BeamRelNode parseQuery(String query) throws ParseException {
```

Comment:
I wouldn't really call this `parseQuery`. This does physical planning, right?

Reply:
I think that it would make sense to refactor (maybe later) so that the pluggable part is `String` -> `RelNode` which might allow plugging things like Calcite babel and then separately `RelNode` -> `BeamRelNode` would be the physical planner.